hermit-sys/build.rs

```diff
@@ -6,17 +7,31 @@ use std::io::Write;
 use std::path::{Path, PathBuf};
 use std::process::Command;
 
+use target_build_utils::TargetInfo;
 use walkdir::{DirEntry, WalkDir};
 
 fn build_hermit(src_dir: &Path, target_dir_opt: Option<&Path>) {
+	let target = TargetInfo::new().expect("Could not get target info");
 	let profile = env::var("PROFILE").expect("PROFILE was not set");
 	let mut cmd = Command::new("cargo");
-	cmd.current_dir(src_dir)
-		.arg("build")
-		.arg("-Z")
-		.arg("build-std=core,alloc")
-		.arg("--target")
-		.arg("x86_64-unknown-hermit-kernel");
+
+	if target.target_arch() == "x86_64" {
+		cmd.current_dir(src_dir)
+			.arg("build")
+			.arg("-Z")
+			.arg("build-std=core,alloc")
+			.arg("--target")
+			.arg("x86_64-unknown-hermit-kernel");
```

Comment:
You could as well put the `if-else/match` statement only into the last argument. The others seem to be identical. BTW: with the latest nightlys you could also specify build-std in the  .cargo/config.